build/nodes-and-endpoints/join-mainnet.md

```diff
@@ -336,22 +336,61 @@ git checkout $BINARY_VERSION
 make build
 cp build/persistenceCore ~/.persistenceCore/cosmovisor/upgrades/$BINARY_VERSION/bin
 ```
-
 Now, at the upgrade height, **Cosmovisor** will upgrade swap the binaries.
 
-### Manual Option
+**2. Auto-download option**   
+Above manual option, still needs upgrade version binary to be manually added under cosmovisor/upgrades/upgrade-name/bin directory.
+For clarity, refer below folder layout for daemon home directory:-
+
+![alt text for screen readers](cosmo-visor-folder-layout.jpg)
+
+However, for people who don't need such control and want an automated setup to download binaries on all the validating nodes, there is an **auto-download** option.
+
+**NOTE**: We don't recommend using auto-download because it doesn't verify in advance if a binary is available. If there will be any issue with downloading a binary, the cosmovisor will stop and won't restart an App (which could lead to a chain halt). Therefore, it is every node operator's responsibility to analyze such risks and use cosmovisor as per requirement.
```

Comment:
We don't recommend it for validators right, but others can use it, correct? 

what do you mean by 'the cosmovisor will stop and won't restart an App'? 
Why would that lead to a chain halt? 

Let's try to be very very specific here

Reply:
1) For Question 1 
 Yes, we do not recommend it for validators. Do you think, we need to mention it directly here? 
The cosmovisor doc states one example for usecase as below:-
"However, for people who don't need such control and want an easier setup (maybe they are syncing a non-validating fullnode and want to do little maintenance), there is another option."
We can put the same example, do you agree?

2) Clarifications related to above NOTE (it is taken as is from cosmovisor doc)
Yes, the chain will halt in case cosmovisor fails to download the binaries. The App is the persistenceCore binary here.
When the upgrade proposal is detected, cosmovisor is restarted which shuts down the persistenceCore daemon.
If correct upgrade version binary is downloaded successfully, cosmovisor is restarted with it and it re-launches the persistenceCore daemon (app)

Reply:
1. Yes let's mention explicitly we don't recommend it for validators. And yes we can use the example for non-validating full-nodes. 

2. But the chain will not halt if one single non-validating node fails to download the binaries via cosmovisor... That particular node will halt, but not the chain.